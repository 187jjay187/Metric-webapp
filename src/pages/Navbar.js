// imports

import React from 'react';
import { useNavigate } from 'react-router-dom';
import { BsFillGearFill, BsCaretLeftFill } from 'react-icons/bs';
import worldometersImg from '../images/worldometers.png';

// create a navbar

const Navbar = () => {
  const navigate = useNavigate();

  // return the logo title & settings icon on the homepage
  // return the back button & the settings icon

  return (
    <nav className="header">
      {
        window.location.pathname === '/'
          ? (
            <span><img className="worldometersImg" src={worldometersImg} alt="worldometers" /></span>
          )
          : (
            <button type="button" onClick={() => navigate(-1)}>
              <BsCaretLeftFill />
              Back
            </button>
          )
      }
      <div className="Covid-title"><h1>Coronavirus</h1></div>
      <div className="header-actions"><span><BsFillGearFill /></span></div>
    </nav>
  );
};

// export

export default Navbar;
