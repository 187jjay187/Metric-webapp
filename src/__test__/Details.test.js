// imports

import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import { BrowserRouter as Router } from 'react-router-dom';
import Navbar from '../pages/Navbar';
import HomePage from '../components/Home';
import store from '../redux/configureStore';

// navbar test
describe('nav components render', () => {
  test('navbar is renders', () => {
    const tree = renderer
      .create(<Router><Provider store={store}><Navbar /></Provider></Router>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  // detailPage test
  test('Details page Renders', () => {
    const tree = renderer
      .create(<Router><Provider store={store}><HomePage /></Provider></Router>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
