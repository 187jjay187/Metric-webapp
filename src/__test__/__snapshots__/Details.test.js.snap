// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`nav components render Details page Renders 1`] = `
<div
  className="pages"
>
  <div
    className="area"
  >
    <img
      alt="africa"
      src="africa.png"
    />
    <h3>
      Continent: Africa
    </h3>
    <p>
      Countries: 58
    </p>
  </div>
  <form
    className="form"
  >
    <div
      className="search-bar"
    >
      <input
        className="form-control"
        onChange={[Function]}
        placeholder="Search country"
        type="text"
        value=""
      />
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 512 512"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M304 192v32c0 6.6-5.4 12-12 12h-56v56c0 6.6-5.4 12-12 12h-32c-6.6 0-12-5.4-12-12v-56h-56c-6.6 0-12-5.4-12-12v-32c0-6.6 5.4-12 12-12h56v-56c0-6.6 5.4-12 12-12h32c6.6 0 12 5.4 12 12v56h56c6.6 0 12 5.4 12 12zm201 284.7L476.7 505c-9.4 9.4-24.6 9.4-33.9 0L343 405.3c-4.5-4.5-7-10.6-7-17V372c-35.3 27.6-79.7 44-128 44C93.1 416 0 322.9 0 208S93.1 0 208 0s208 93.1 208 208c0 48.3-16.4 92.7-44 128h16.3c6.4 0 12.5 2.5 17 7l99.7 99.7c9.3 9.4 9.3 24.6 0 34zM344 208c0-75.2-60.8-136-136-136S72 132.8 72 208s60.8 136 136 136 136-60.8 136-136z"
        />
      </svg>
    </div>
  </form>
  <ul
    className="list-country"
  />
</div>
`;

exports[`nav components render navbar is renders 1`] = `
<nav
  className="header"
>
  <span>
    <img
      alt="worldometers"
      className="worldometersImg"
      src="worldometers.png"
    />
  </span>
  <div
    className="Covid-title"
  >
    <h1>
      Coronavirus
    </h1>
  </div>
  <div
    className="header-actions"
  >
    <span>
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 16 16"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M9.405 1.05c-.413-1.4-2.397-1.4-2.81 0l-.1.34a1.464 1.464 0 0 1-2.105.872l-.31-.17c-1.283-.698-2.686.705-1.987 1.987l.169.311c.446.82.023 1.841-.872 2.105l-.34.1c-1.4.413-1.4 2.397 0 2.81l.34.1a1.464 1.464 0 0 1 .872 2.105l-.17.31c-.698 1.283.705 2.686 1.987 1.987l.311-.169a1.464 1.464 0 0 1 2.105.872l.1.34c.413 1.4 2.397 1.4 2.81 0l.1-.34a1.464 1.464 0 0 1 2.105-.872l.31.17c1.283.698 2.686-.705 1.987-1.987l-.169-.311a1.464 1.464 0 0 1 .872-2.105l.34-.1c1.4-.413 1.4-2.397 0-2.81l-.34-.1a1.464 1.464 0 0 1-.872-2.105l.17-.31c.698-1.283-.705-2.686-1.987-1.987l-.311.169a1.464 1.464 0 0 1-2.105-.872l-.1-.34zM8 10.93a2.929 2.929 0 1 1 0-5.86 2.929 2.929 0 0 1 0 5.858z"
        />
      </svg>
    </span>
  </div>
</nav>
`;
